flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/table/restrictions/BigQueryRestriction.java

```diff
@@ -0,0 +1,274 @@
+                                    }
+                                }
+
+                                return Optional.empty();
+                            });
+        }
+
+        return Optional.empty();
+    }
+
+    private static Optional<String> convertLogicExpressionWithOperandsSymbol(
+            String operandsSymbol, CallExpression call) {
+        List<ResolvedExpression> args = call.getResolvedChildren();
+        if (args == null || args.size() != 2) {
+            return Optional.empty();
+        }
+
+        Optional<String> left = convert(args.get(0));
+        Optional<String> right = convert(args.get(1));
+        if (left.isPresent() && right.isPresent()) {
+            return Optional.of(
+                    String.format("(%s %s %s)", left.get(), operandsSymbol, right.get()));
+        }
+
+        return Optional.empty();
+    }
+
+    private static String addSingleQuotes(String input) {
+        return "'" + input + "'";
+    }
+
+    private static Optional<Object> convertLiteral(ValueLiteralExpression expression) {
+        Optional<?> value =
+                expression.getValueAs(
+                        expression.getOutputDataType().getLogicalType().getDefaultConversion());
+        return value.map(
+                o -> {
+                    if (o instanceof LocalDateTime) {
+                        return addSingleQuotes(((LocalDateTime) o).toString());
+                    } else if (o instanceof Instant) {
+                        return addSingleQuotes(((Instant) o).toString());
+                    } else if (o instanceof LocalTime) {
+                        return addSingleQuotes(((LocalTime) o).toString());
+                    } else if (o instanceof LocalDate) {
+                        return addSingleQuotes(((LocalDate) o).toString());
+                    } else if (o instanceof String) {
+                        return addSingleQuotes((String) o);
+                    }
+
+                    return o;
+                });
+    }
+
+    private static Optional<String> convertOperationPartsWithItsSymbol(
+            String operationSymbol, CallExpression call) {
+        List<ResolvedExpression> args = call.getResolvedChildren();
+        if (args.size() != 2) {
+            return Optional.empty();
+        }
+
+        Expression left = args.get(0);
+        Expression right = args.get(1);
+
+        if (left instanceof FieldReferenceExpression && right instanceof ValueLiteralExpression) {
+            String name = ((FieldReferenceExpression) left).getName();
+            Optional<Object> lit = convertLiteral((ValueLiteralExpression) right);
+            if (lit.isPresent()) {
+                return Optional.of(String.format("(%s %s %s)", name, operationSymbol, lit.get()));
+            }
+        } else if (left instanceof ValueLiteralExpression
+                && right instanceof FieldReferenceExpression) {
```

Comment:
Is it possible to have left, right both as instance of the same thing? Like both ValueLiteralExpression, or FieldReferenceExpression?
In any case, a suggestion is to convert left and right independently into a common type `Optional<Object>` and then if both are present, we simply convert with `(leftOption.get(), operationSymbol, rightOption.get())`

Reply:
Applied your suggestion.

`Is it possible to have left, right both as instance of the same thing? Like both ValueLiteralExpression, or FieldReferenceExpression?`
That should not happen. The filter expressions are resolved and validated by Flink by reaching this step.